### DISCUSSION TITLE
Adding a gpg key isn't working

### DISCUSSION THREAD

**Discussion by bcl** [community/community#22864] [New to GitHub]
<p>When I try to add a gpg key to my account I get this error:</p>
<p>“keyring packet should only contain a single entity”</p>
<p>I’ve tried with both my personal key and my work key and get the same results.</p>
<p>You can see the key here - <a href="https://keys.fedoraproject.org/pks/lookup?search=0xD29845A70F5017DE&amp;op=vindex" rel="nofollow noopener">https://keys.fedoraproject.org/pks/lookup?search=0xD29845A70F5017DE&amp;op=vindex</a></p>

**Comment by bcl** [comment]
<p>And oddly enough when I paste the key from the keyserver instead of the one exported locally it works.</p>

**Comment by jaytelford** [comment]
<p>I had this probelm when trying to copy/paste keys created on Windows 10.</p>
<p>Sometimes exported keys get unwanted or unintended formatting.  Its best just to <em>cat out</em> the key and copy paste it directly from the Terminal.  That way (unless you have some custome setup for your terminal output) you can be sure that you are copying just the key in plain text, without any extra formatting or anything.</p>
<p>If you can’t cat the key and copy directly from Terminal, then I would suggest copy pasting your key into a text editor that is specifically designed for handling code.  This will make sure that they key is pasted in plain text and it will allow you remove any unwanted formatting.</p>
<p>If you don’t have a native program, you can use pastebin - you will just have to remember not to actually save your key, or at the very least delete it aftewards.</p>
<p>Its always probably a good idea to note that although my problem was created by Windows 10 - any system, even a Linux based one, can add weird formatting or characters to your key if you try opening it with native programs instead of directly copying from Terminal.</p>

**Comment by seveas** [comment, marked as answer]
<p><a class="mention" href="/bcl">@bcl</a> we’ve seen this before: the key probably exists twice in your local keychain and the export command copies both. The GitHub importer only sees that there are two keys, not that they are identical. Here’s your key after importing into a fresh keychain and exporting it, try this one.</p>
<p><a href="https://gist.github.com/seveas/89d5e3cf1aaa313fe042909c02849b96" rel="nofollow noopener">https://gist.github.com/seveas/89d5e3cf1aaa313fe042909c02849b96</a></p>

**Comment by bcl** [comment]
<p>Ah, that could be. I was able to successfully import the one available from the Fedora keyserver, thanks!</p>

**Comment by BenSturmfels** [comment]
<p>The solution for me was specifying the key with the ID rather than the email address (as I have multiple keys with the same email address). For example:</p>
<pre><code>$ gpg --list-secret-keys
$ gpg --export --armor --export-options export-minimal E2C9C068023C05F0 &lt;-- Your key ID here
</code></pre>

**Reply by Sanix-Darker** [reply]
yeah, this solution works !

**Reply by saurabh-mahajan** [reply]
ya its working for me to...
Thanks alot !

**Reply by Cosmin92RO** [reply]
Thank you for solution!

**Reply by kaikaibenkai** [reply]
It works!

**Reply by chawatphonr** [reply]
Thanks!
